claim-check/code-samples/sample-4/client-producer/Program.cs

```diff
@@ -0,0 +1,115 @@
+            Console.WriteLine("Done.");
+            Console.WriteLine();
+
+            Console.WriteLine("Press any key to exit the sample application.");
+            Console.ReadLine();
+        }
+
+        private static void SendNotification()
+        {
+            string brokerList = ConfigurationManager.AppSettings["EH_FQDN"];
+            string connectionString = ConfigurationManager.AppSettings["EH_CONNECTION_STRING"];
+            string topic = ConfigurationManager.AppSettings["EH_NAME"];
+            string caCertLocation = ConfigurationManager.AppSettings["CA_CERT_LOCATION"];
+
+            Worker.Producer(brokerList, connectionString, topic, caCertLocation, payloadDetails).Wait();
+        }
+
+        private static async Task ProcessAsync()
+        {
+            // Initialize variables
+            CloudStorageAccount storageAccount = null;
+            CloudBlobContainer cloudBlobContainer = null;
+            string sourceFile = null;
+
+            // Store message in blob storage
+            string storageConnectionString = ConfigurationManager.AppSettings["STORAGE_CONNECTION_STRING"];
+            
+            // Check whether the connection string can be parsed.
+            if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
+            {
+                try
+                {
+                    // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+                    CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+
+                    // Create a container called 'heavypayload' and append a GUID value to it to make the name unique. 
+                    var containerName = "heavypayload" + Guid.NewGuid().ToString();
+                    Console.WriteLine("- Creating container '{0}'", containerName);
+                    cloudBlobContainer = cloudBlobClient.GetContainerReference(containerName);
+                    await cloudBlobContainer.CreateAsync();
+
+                    // Set the permissions so the blobs are public. 
+                    BlobContainerPermissions permissions = new BlobContainerPermissions
+                    {
+                        PublicAccess = BlobContainerPublicAccessType.Blob
+                    };
+                    await cloudBlobContainer.SetPermissionsAsync(permissions);
+
+                    // This would ideally be a large message/payload/file
+                    // Create a file in your local MyDocuments folder to upload to a blob.
+                    string localPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+                    string localFileName = "HeavyPayload_" + Guid.NewGuid().ToString() + ".txt";
+                    sourceFile = Path.Combine(localPath, localFileName);
+                    // Write contents to file
+                    File.WriteAllText(sourceFile, "Hello, World! This is a huge file");
```

Comment:
Use File.WriteAllTextAsync()

Reply:
I don't have that method

Reply:
Correction, found the method and used the Async version

Reply:
It doesn't always show up in Intellisense in VS Code, but it is there.  As an alternative, instead of putting files into the Documents folder, we can just upload the text directly.  Something like below would work.

```csharp
var blobName = "HeavyPayload_" + Guid.NewGuid().ToString() + ".txt";
Console.WriteLine("- Uploading to Blob storage as blob '{0}'", blobName);

// Get a reference to the blob address, then upload the file to the blob.
// Use the value of localFileName for the blob name.
CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(blobName);

// This would ideally be a large message/payload/file
await cloudBlockBlob.UploadTextAsync("Hello, World! This is a huge file");

// Store details of blob to be sent as a message
return new PayloadDetails
{
    ContainerName = cloudBlobContainer.Name,
    BlobName = blobName
};
```